dora/core/server/worker/src/main/java/alluxio/worker/dora/DoraMetaManager.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.collections.LockPool;
+import alluxio.concurrent.LockMode;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.proto.meta.DoraMeta;
+import alluxio.proto.meta.DoraMeta.FileStatus;
+import alluxio.underfs.Fingerprint;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * The Dora metadata manager that orchestrates the metadata operations and provides the
+ * concurrency control.
+ */
+public class DoraMetaManager {
+  private final DoraMetaStore mMetastore;
+  private final CacheManager mCacheManager;
+  private final PagedDoraWorker mDoraWorker;
+  private final UnderFileSystem mUfs;
+  private final LockPool<String> mPathLocks;
```

Comment:
so this LockPool is a LockPool of String. I feel there might be 2 alternatives
1. Use a LockPool of long and the key of the lock pool is the hash of the full ufs path name. This helps us save memory. 
2. Use segmented lock (an example can be found in https://github.com/Alluxio/alluxio/blob/main/dora/core/client/fs/src/main/java/alluxio/client/file/cache/cuckoofilter/SegmentedLock.java). Basically let all paths to share a set of locks (say 10k). Because in Dora, the path lock is more a latch than a lock and it is only used to ensure the consistency between metadata cache/rocksdb and  file cache, the lock is very lightweight and is acquired/released very quickly. So i think it's okay to let different paths to share the same lock. In alluxio 2.x this is not feasible because the many things such as fetching status from ufs can happen after acquiring a lock.

Reply:
We might need collect some benchmark and flamechart for the lockpool

Reply:
discussed with @beinan offline, we are going to no add the lock pool for now